src/app/utils/utils.ts

```diff
@@ -175,4 +175,11 @@ export function removeColorUnderPercent(percent: number, usage: Map<string, numb
             e.enabled = false;
         });
     });
+}
+
+export function getPaletteEntryByColorRef(palettes: Palette[], ref: string): PaletteEntry {
```

Comment:
I do understand the use here, mostly due to because usage is not using PaletteEntry object but only the ref. Could be remove at a later time when usage is using PaletteEntry. Which should greatly increase perf as well since no lookup will be required for each bead.

Reply:
Sure, I looked into attempting that but the with the usage map being used in a number of places I was a little hesitant to change it. I'm happy to change try and update it if performance is a concern at this stage.